debezium-core/src/main/java/io/debezium/pipeline/signal/FileSignalThread.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.pipeline.signal;
+
+import java.io.FileWriter;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.DebeziumException;
+import io.debezium.config.CommonConnectorConfig;
+import io.debezium.config.Configuration;
+import io.debezium.config.Field;
+import io.debezium.connector.common.RelationalBaseSourceConnector;
+import io.debezium.pipeline.source.snapshot.incremental.AbstractReadOnlyIncrementalSnapshotChangeEventSource;
+import io.debezium.schema.DataCollectionId;
+
+/**
+ * The class responsible for processing of signals delivered to Debezium via a file.
+ * The signal message must have the following structure:
+ * <ul>
+ * <li>{@code data STRING} - the data in JSON format that are passed to the signal code
```

Comment:
this is what i thought for file based signals. without `id` and `type`. should it be with`id`,`type`,`data`? comma separated?

```
{"data-collections": ["public.MyFirstTable", "public.MySecondTable"]}
{"data-collections": ["public.MyFirstTable", "public.MySecondTable2"]}
{"data-collections": ["public.MyFirstTable", "public.MySecondTable3"]}
{"data-collections": ["public.MyFirstTable", "public.MySecondTable4"]}
```